controller/lib/core/controller.h

```diff
@@ -6,15 +6,21 @@
 #include "network_protocol.pb.h"
 #include "pid.h"
 #include "units.h"
+#include <utility>
+
+struct ControllerState {
+  // True if this is the first breath of a new cycle.
+  bool is_new_breath;
```

Comment:
I'm worried about including this in the "state" because, unlike ActuatorsState, this variable cannot be subsampled (or a communication packet lost), or you'll miss the beginning of a breath.

I think it's an actual problem because we will want to communicate breath boundaries to the GUI, so GUI can compute breath-based alarms, and GUI may want to subsample controller state history for performance reasons.

One way you could address this is by changing it to num_breaths which can be subsampled.

Reply:
This is an interesting question...

For the GUI I totally understand sending a count.  But for this struct, which is not the proto, if the consumers of it want to know "is this a new breath", then if we put a count in here then the consumers are going to have to keep track of the last-seen-count.  Which seems error-prone?

WDYT?

Reply:
Forgot to respond to this one. If this isn't communicated to the GUI, then it's fine for now. It'll get refactored away anyway when we make all this sensor correction business happen in Run().